rhino/src/main/java/org/mozilla/javascript/NativeObject.java

```diff
@@ -722,22 +722,42 @@ private static Object js_assign(
                 continue;
             }
             Scriptable sourceObj = ScriptRuntime.toObject(cx, scope, args[i]);
-            Object[] ids = sourceObj.getIds();
+            Object[] ids;
+            if (sourceObj instanceof ScriptableObject) {
+                ids = ((ScriptableObject) sourceObj).getIds(false, true);
+            } else {
+                ids = sourceObj.getIds();
+            }
             for (Object key : ids) {
                 if (key instanceof Integer) {
                     int intId = (Integer) key;
                     if (sourceObj.has(intId, sourceObj) && isEnumerable(intId, sourceObj)) {
                         Object val = sourceObj.get(intId, sourceObj);
                         AbstractEcmaObjectOperations.put(cx, targetObj, intId, val, true);
                     }
-                } else {
+                } else if (key instanceof String) {
                     String stringId = ScriptRuntime.toString(key);
                     if (sourceObj.has(stringId, sourceObj) && isEnumerable(stringId, sourceObj)) {
                         Object val = sourceObj.get(stringId, sourceObj);
                         AbstractEcmaObjectOperations.put(cx, targetObj, stringId, val, true);
                     }
                 }
             }
+
+            // This is a separate loop for Symbols, as they must be
+            // copied over after string properties
+            if (sourceObj instanceof ScriptableObject) {
+                for (Object key : ids) {
+                    if (key instanceof Symbol) {
```

Comment:
is this correct here or should we use ScriptRuntime.isSymbol(key)

Reply:
I compared this to node to conclude that it is correct. I've added a couple of test cases now.